darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,374 @@
+            # since the figures are anchored on the left hand
+            # side, the appropriate x-value is the difference
+            # of the page midpoint and half the figure width
+            fig_x = page_midpoint - (fig_w / 2)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            # save the figure in a temporary directory
+            fig_savepath = os.path.join(tmpdirname, "fig.png")
+            fig.savefig(fig_savepath, dpi=300)
+            # add the saved image to the PDF at the
+            # desired location and dimensions
+            self.image(name=fig_savepath, x=fig_x, y=fig_y, w=fig_w)
+
+
+def setup_parser(parser):
+    """
+    Configures the command line arguments.
+
+    Parameters
+    ----------
+    parser : ArgumentParser object
+        Command line argument parser.
+    """
+    parser.add_argument(
+        "log_path",
+        type=str,
+        help="Specify path to darshan log.",
+    )
+
+
+def main(args=None):
+    """
+    Generates a PDF report containing a summary
+    for an input Darshan log file.
+    """
+    if args is None:
+        # if no arguments are input, use setup_parser()
+        parser = argparse.ArgumentParser()
+        setup_parser(parser=parser)
+        args = parser.parse_args()
+
+    log_path = args.log_path
+    # get the filename from the log path
+    log_filename = os.path.basename(log_path)
+    # load the darshan report
+    report = darshan.DarshanReport(log_path, read_all=True)
+    # create the PDF object in portrait orientation, letter
+    # dimensions (8.5 x 11"), and using point units (72 points/inch)
+    pdf = PDF(
+        orientation="portrait",
+        unit="pt",
+        format="letter",
+        metadata_dict=report.data["metadata"],
+    )
+    # add the first page
+    pdf.add_page()
+    # set the title
+    pdf.set_title(log_filename=log_filename)
+    # add the summary table at the top of page 1 that contains
+    # the jobid/uid/nprocs/runtime info
+    pdf.add_metadata_table()
+    # construct dictionaries of the arguments for each figure/function
+    hmap_func_args = {"mods": ["DXT_POSIX"], "ops": ["read", "write"], "xbins": 200}
+    opct_func_args = {"ax": None}
+    # pick the figure widths
+    fig_w = 400
+    # enable the darshan experimental module
+    darshan.enable_experimental()
+    # add the figures to the PDF
+    pdf.add_figure(
+        log_path=log_path,
+        plotting_func=plot_heatmap,
+        func_args=hmap_func_args,
+        fig_y=100,
+        fig_w=fig_w,
+        center=True,
+    )
+    pdf.add_figure(
+        log_path=log_path,
+        plotting_func=plot_opcounts,
+        func_args=opct_func_args,
+        fig_y=410,
+        fig_w=fig_w,
+        center=True,
+        report=report,
+    )
+    # save the PDF at the desired location with the filename
+    fname_noext = os.path.splitext(log_filename)[0]
+    output_filename = os.getcwd() + f"/report_{fname_noext}.pdf"
```

Comment:
`os.path.join()` is generally more robust than raw string concatenation

Reply:
Absolutely, I'll make sure to change that.

Reply:
Current version uses `os.path.join()`.